File: implementation/scipamato/public/web/src/main/java/ch/difty/scipamato/publ/web/config/WicketWebSecurityConfig.java
package ch.difty.scipamato.publ.web.config;

import org.springframework.boot.actuate.autoconfigure.security.servlet.EndpointRequest;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;
import org.springframework.security.authentication.AuthenticationManager;
import org.springframework.security.config.annotation.web.builders.HttpSecurity;
import org.springframework.security.config.annotation.web.configuration.WebSecurityConfigurerAdapter;
import org.springframework.security.core.userdetails.User;
import org.springframework.security.core.userdetails.UserDetailsService;
import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;
import org.springframework.security.crypto.password.PasswordEncoder;
import org.springframework.security.provisioning.InMemoryUserDetailsManager;

import ch.difty.scipamato.publ.config.ScipamatoPublicProperties;

@Configuration
public class WicketWebSecurityConfig extends WebSecurityConfigurerAdapter {

    private static final String ADMIN_ROLE = "ADMIN";

    private final ScipamatoPublicProperties properties;

    WicketWebSecurityConfig(final ScipamatoPublicProperties properties) {
        this.properties = properties;
    }

    @Bean(name = "authenticationManager")
    @Override
    public AuthenticationManager authenticationManagerBean() throws Exception {
        return super.authenticationManagerBean();
    }

    @Override
    protected void configure(HttpSecurity http) throws Exception {
        // @formatter:off
        http.csrf().disable()
            .headers().frameOptions().disable()
            .and()
            .authorizeRequests()
                .requestMatchers(EndpointRequest.to("health", "info")).permitAll()
                .antMatchers("/actuator/").hasRole(ADMIN_ROLE)
                       .requestMatchers(EndpointRequest.toAnyEndpoint()).hasRole(ADMIN_ROLE)
                .antMatchers("/**").permitAll()
            .and()
                .logout().permitAll();
        // @formatter:on
    }

    @Bean
    public static PasswordEncoder passwordEncoder() {
        return new BCryptPasswordEncoder();
    }

    @Override
    @Bean
    public UserDetailsService userDetailsService() {
        InMemoryUserDetailsManager manager = new InMemoryUserDetailsManager();
        manager.createUser(User
            .withUsername(properties.getManagementUserName())
            .password(passwordEncoder().encode(properties.getManagementUserPassword()))
            .roles(ADMIN_ROLE)
            .build());
        return manager;
    }

}

File: public/public-web/src/main/kotlin/ch/difty/scipamato/publ/web/config/WicketWebSecurityConfig.kt
package ch.difty.scipamato.publ.web.config

import ch.difty.scipamato.publ.config.ScipamatoPublicProperties
import org.springframework.boot.actuate.autoconfigure.security.servlet.EndpointRequest
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration
import org.springframework.security.authentication.AuthenticationManager
import org.springframework.security.config.annotation.web.builders.HttpSecurity
import org.springframework.security.config.annotation.web.configuration.WebSecurityConfigurerAdapter
import org.springframework.security.core.userdetails.User
import org.springframework.security.core.userdetails.UserDetailsService
import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder
import org.springframework.security.crypto.password.PasswordEncoder
import org.springframework.security.provisioning.InMemoryUserDetailsManager

@Configuration
open class WicketWebSecurityConfig(
    private val properties: ScipamatoPublicProperties,
) : WebSecurityConfigurerAdapter() {

    @Bean(name = ["authenticationManager"])
    @Throws(Exception::class)
    override fun authenticationManagerBean(): AuthenticationManager = super.authenticationManagerBean()

    @Throws(Exception::class)
    override fun configure(http: HttpSecurity) {
        http.csrf().disable()
            .headers().frameOptions().disable()
            .and()
            .authorizeRequests { r ->
                r.requestMatchers(EndpointRequest.to("health", "info")).permitAll()
                    .antMatchers("/actuator/").hasRole(ADMIN_ROLE)
                    .requestMatchers(EndpointRequest.toAnyEndpoint()).hasRole(ADMIN_ROLE)
                    .antMatchers("/**").permitAll()
            }
            .logout { logout -> logout.permitAll() }
    }

    @Bean
    public override fun userDetailsService(): UserDetailsService =
        InMemoryUserDetailsManager().apply {
            createUser(User
                .withUsername(properties.managementUserName)
                .password(passwordEncoder().encode(properties.managementUserPassword))
                .roles(ADMIN_ROLE)
                .build())
        }

    companion object {
        private const val ADMIN_ROLE = "ADMIN"

        @Bean
        fun passwordEncoder(): PasswordEncoder = BCryptPasswordEncoder()
    }
}
